photon-ml/src/main/scala/com/linkedin/photon/ml/function/ValueAndGradientAggregator.scala

```diff
@@ -15,24 +15,21 @@
 package com.linkedin.photon.ml.function
 
 import breeze.linalg.{DenseVector, Vector, axpy}
-import com.linkedin.photon.ml.data.{LabeledPoint, ObjectProvider}
-import com.linkedin.photon.ml.normalization.NormalizationContext
 import org.apache.spark.broadcast.Broadcast
 import org.apache.spark.rdd.RDD
 
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.function.glm.PointwiseLossFunction
+import com.linkedin.photon.ml.normalization.NormalizationContext
 
+// TODO: Better document this algorithm, especially normalization.
```

Comment:
I wrote a wiki page for the algorithm at LinkedIn wiki. @ashelkovnykov do you know what I'm talking about? Since it's open sourced I cannot use the wiki link here. I don't know how to reconcile this.